internal/client/runner.go

```diff
@@ -112,8 +112,21 @@ func remoteOpPreferred(ctx context.Context, client pb.WaypointClient, project *p
 		return true, nil
 	}
 
-	// Check to see if we have a global default ODR profile
+	// For now, if any app has an ODR profile set, we'll prefer remote for every op
+	// NOTE: this means that it isn't possible to have one app in a project execute
+	// locally only, and another execute remotely.
+	// TODO(izaak): it's possible for us to fix this by invoking this once per app, instead of once per project
+	for _, runnerCfg := range runnerCfgs {
+		if runnerCfg == nil {
+			continue
+		}
+		if runnerCfg.Profile != "" {
+			log.Debug("At least one app has an explicit ODR profile set - choosing remote operations for all app operations.")
```

Comment:
Why a warn instead of debug like the other log messages?

Reply:
I think Warn is good here mostly because this could be a surprising behavior if they intentionally configure some applications to be remote and some not and it switches on them. So I see `Warn` as like, a heads up.